src/tini.c

```diff
@@ -133,8 +133,19 @@ int spawn(const signal_configuration_t* const sigconf_ptr, char* const argv[], i
 		}
 
 		execvp(argv[0], argv);
-		PRINT_FATAL("Executing child process '%s' failed: '%s'", argv[0], strerror(errno));
-		return 1;
+	    	// execvp will only return on an error so make sure that we check the errno
+		// and exit with the correct return status for the error that we encountered
+		int status = EXIT_FAILURE;
+		switch errno {
+		case ENOENT:
+			status = 127;
+			break;
+		case EACCES:
+			status = 126;
+			break;
+		}
+		fprintf(stderr, "%s\n", strerror(errno));
```

Comment:
Ya, this is the hard part.  

I actually prefer things like this to be opaque as pass though as much as they can without modification.  For higher levels it makes parsing very hard if every init added their name in the log messages when its more of an implementation detail to the end users.

We can add more info like you suggested or `exec %s: %s`(arg0, errno) or something similar but not tied to an implementation.  Making it more anonymous is my preference, not to take away from the underlying tech that is being used but because it's hard on users to parse output when its implementation dependent.  

I've always tried to build the runtimes where people feel like they are not interacting with a container but a binary and the container is just an implementation detail of how its run; they shouldn't notice a difference.  I think this is a good model to follow for an init as well but that is my general thinking behind this. 

Let me know what you think.